Integrations/integration-OpsGenie.yml

```diff
@@ -0,0 +1,141 @@
+                Headers: {
+                    Accept: ['application/json'],
+                    Authorization: ['GenieKey ' + params.apiKey]
+                }
+            },
+            false,
+            params.proxy
+        );
+
+        if (res.StatusCode < 200 || res.StatusCode >= 300) {
+            logError('OpsGenie error: Request failed for ' + requestUrl + '. Returned body is ' + JSON.stringify(res));
+            throw 'OpsGenie: Request Failed.\nStatus code: ' + res.StatusCode + '.\nBody: ' + JSON.stringify(res) + '.';
+        }
+
+        var jsonBody = '';
+        try {
+            jsonBody = JSON.parse(res.Body);
+        } catch (err) {
+            logInfo('OpsGenie error: Could not parse respons for ' + requestUrl + '. Error is ' + err + '. Returned body is ' + JSON.stringify(res));
+            throw 'OpsGenie error\n Could not parse respons for ' + requestUrl + '\n Error is ' + err;
+        }
+        return jsonBody;
+    };
+
+    var getOnCall = function(schedule) {
+        var url = serviceURLs.schedules + '/' + encodeURIComponent(schedule) + '/on-calls?scheduleIdentifierType=name';
+        var res = sendRequest(url);
+        var md = '### OpsGenie On-Call Schedule __' + schedule + '__\n';
+        var ec = {};
+        if (res && res.data && res.data.onCallParticipants && res.data.onCallParticipants.length > 0) {
+            var onCall = dq(res.data.onCallParticipants,'name')
+            md += 'Currently on-call for __' + schedule + '__ schedule:\n';
+            ec.OpsGenie = {Schedule: schedule, OnCall: []};
+            onCall.forEach(function(name) {
+                var user = getUser(name);
+                md += '- ' + user.Contents.fullName + ' ('+name+')\n';
+                ec.OpsGenie.OnCall.push({email: name, name:user.Contents.fullName})
+            });
+        } else {
+            md += 'Cannot find on-call for __' + schedule + '__\n';
+        }
+        return {Type: entryTypes.note, Contents: res.data.onCallParticipants, EntryContext: ec, ContentsFormat: formats.json, HumanReadable: md};
+    };
+
+    var getUser = function(userID) {
+        var url = serviceURLs.users + '/' + encodeURIComponent(userID);
+        var res = sendRequest(url);
+        var md = '### OpsGenie User Info\n';
+        var ec = {};
+        if (res && res.data && res.data) {
+            var user = res.data;
+            md += objToMd(user);
+        } else {
+            md = 'Cannot find user ' + userID + '\n';
+        }
+        return {Type: entryTypes.note, Contents: res.data, EntryContext: ec, ContentsFormat: formats.json, HumanReadable: md};
+    };
+
+    switch (command) {
+        case 'test-module':
+            var url = serviceURLs.account;
+            sendRequest(url);
+            return 'ok';
+
+        case 'opsgenie-get-on-call':
+            return getOnCall(args.schedule);
+
+        case 'opsgenie-get-user':
+            return getUser(args.userID);
+
+        default:
```

Comment:
can be removed, can't happen

Reply:
I think it's better to keep so if someone adds a command without adding the case, it's easier to see.